src/goose/brokers/redis/batch.clj

```diff
@@ -0,0 +1,25 @@
+(ns goose.brokers.redis.batch
+  (:require [goose.brokers.redis.commands :as redis-cmds]
+            [goose.defaults :as d]
+            [taoensso.carmine :as car]))
+
+(defn- set-batch-state
+  [{:keys [id callback-fn-sym jobs]}]
+  (let [batch (d/prefix-batch id)
+        enqueued-jobs-set (d/suffix-batch-set id d/enqueued-job-set)]
+    (car/hmset batch
+               (name :id) id
```

Comment:
I've used the `name` fn here so that I don't have to hard code the field names. 
However, this doesn't seem like a better alternative either, looking for suggestions on improving this

Reply:
Updated the implementation of the fn to use the `hmset` Carmine fn, which takes a map.

Additionally, you'd define the keys in the batch state as, `batch-state (select-keys batch [:id :callback-fn-sym])`